docs/docs/genai/prompt-registry/adapt-prompts.mdx

```diff
@@ -0,0 +1,361 @@
+- **Switching Providers**: Changing from one LLM provider to another (e.g., OpenAI to Anthropic)
+
+## Quick Start
+
+Here's a simple example of rewriting prompts when switching models:
+
+```python
+import mlflow
+import openai
+from mlflow.genai.optimize import LLMParams
+
+# Register your current prompt
+prompt = mlflow.genai.register_prompt(
+    name="qa",
+    template="Answer the following question: {{question}}",
+)
+
+
+# Your current function
+def predict_fn(question: str) -> str:
+    completion = openai.OpenAI().chat.completions.create(
+        model="gpt-4o-mini",  # Target model
+        messages=[{"role": "user", "content": prompt.format(question=question)}],
+    )
+    return completion.choices[0].message.content
+
+
+# Example data showing desired outputs (e.g., outputs produced by gpt-4o)
+dataset = [
+    {"inputs": {"question": "What is the capital of France?"}, "outputs": "Paris"},
+    {"inputs": {"question": "What is the capital of Germany?"}, "outputs": "Berlin"},
+    {"inputs": {"question": "What is the capital of Italy?"}, "outputs": "Rome"},
+    {"inputs": {"question": "What is the capital of Spain?"}, "outputs": "Madrid"},
+    {"inputs": {"question": "What is the capital of Portugal?"}, "outputs": "Lisbon"},
+    {"inputs": {"question": "What is the capital of Switzerland?"}, "outputs": "Bern"},
+    {"inputs": {"question": "What is the capital of Austria?"}, "outputs": "Vienna"},
+    {"inputs": {"question": "What is the capital of Belgium?"}, "outputs": "Brussels"},
+    {
+        "inputs": {"question": "What is the capital of Netherlands?"},
+        "outputs": "Amsterdam",
+    },
+    {"inputs": {"question": "What is the capital of Norway?"}, "outputs": "Oslo"},
+    {"inputs": {"question": "What is the capital of Sweden?"}, "outputs": "Stockholm"},
+]
+
+# Rewrite prompts for a new model
+result = mlflow.genai.adapt_prompts(
+    predict_fn=predict_fn,
+    train_data=dataset,
+    target_prompt_uris=[prompt.uri],
+    optimizer_lm_params=LLMParams(model_name="openai:/gpt-4o"),
+)
+
+# Use the optimized prompt
+optimized_prompt = result.optimized_prompts[0]
+print(f"## Optimized template: {optimized_prompt.template}")
+```
+
+In this example, this API produces a more informative prompt template:
+
+```
+## Optimized template:
+Please provide a concise response to the question enclosed in the {{question}} variable.
+Ensure that the answer addresses the main question directly.
+For instance, if a location is asked about, mention only the primary information required, such as the name of the capital city for a country-related question.
+Avoid additional context, complete sentences, or extraneous information in your response.
+Your aim is to deliver a precise, correct answer that matches the expectation outlined.
+```
+
+## Components
+
+### 1. Predict Function
+
+Your `predict_fn` should:
+
+- Accept inputs as keyword arguments
```

Comment:
Is this optional or required? `should` reads as optional. If it's required, let's use `must` to remove ambiguity.